### ISSUE TITLE
`AttributeError: 'MaskedNDArray' object has no attribute 'value'` when creating Pyriod object

### ISSUE THREAD

**Issue by warrickball** [keatonb/Pyriod#7]
I thought I'd try out Pyriod but am running into issues. The following short script (in a Python script or Jupyter notebook)
```python
import lightkurve as lk
from Pyriod import Pyriod

lc = lk.search_lightcurve('SX Phe', mission='TESS', exptime=120).download_all()[0].remove_nans()
pyriod = Pyriod(lc)
```
produces the error
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_497628/1489146282.py in <module>
      3 
      4 lc = lk.search_lightcurve('SX Phe', mission='TESS', exptime=120).download_all()[0].remove_nans()
----> 5 pyriod = Pyriod(lc)

~/.local/lib/python3.10/site-packages/Pyriod/Pyriod.py in __init__(self, lc, amp_unit, freq_unit, gui, **kwargs)
    229 
    230         #And keep track of residuals time series
--> 231         self.lc["resid"] = self.lc["flux"].value - self.lc["model"].value
    232 
    233 

AttributeError: 'MaskedNDArray' object has no attribute 'value'
```
It appears that `lc['model']` is the source of the problem. I can assign new columns to `lc` using something like `lc['resid'] = lc['flux'].value*1` without any errors.

This might also be because I'm using Astropy 5.0 because I end up in an endless loop of failed builds for anything <5.0.

Incidentally, this is Python 3.10.4 on Fedora 35. Not sure which package versions are important.

**Comment by keatonb** [comment]
Hi Warrick, this is a conflict with Astropy 5.0 that I've been meaning to get around to resolving.  I'll close this issue when I do, thanks!

**Comment by warrickball** [comment]
Let me know if I can potentially help. I'm not that familiar with Astropy's data structures or what's changed in 5.0 but I can try to fix it anyway.

I managed to do some of the analysis I wanted by using the live notebook in the [TESS SciCon 2 poster](https://zenodo.org/record/5129684), which worked like a charm!

**Comment by warrickball** [comment]
_Edit: I'd written about Jupyter woes that I've fixed._

I think I have something that solves my original problem but it seems to break the fitting processing because the initial amplitudes of the peaks in the periodogram seem to be all wrong, probably because of something unit-related.

My "fix" was to leave the units in the initial model and residual columns that are added during `__init__`. i.e.
```python
         #Store version sampled as the data as lc column
         initmodel = np.zeros(len(self.lc))+np.mean(self.lc.flux[self.include])
```
and
```python
         #And keep track of residuals time series
         self.lc["resid"] = self.lc["flux"] - self.lc["model"]
```
but, though I can at least start the session, that seems to cause confusion once the widgets are actually running and fitting data.

**Comment by keatonb** [comment]
I believe I sorted this out with v0.1.11.  Your comments were very helpful!  Would appreciate you letting me know if this working for you now with astropy>5.  Cheers!

**Comment by warrickball** [comment]
I think my original issue is resolved but now when I try the "Refine fit" button in the Periodogram window, I get the error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/.local/lib/python3.10/site-packages/Pyriod/Pyriod.py:917, in Pyriod.fit_model(self, *args)
    914     self.log("Fit properties:"+self.fit_result.fit_report())
    915     self._update_values_from_fit(self.fit_result.params,prefixmap)
--> 917 self._update_lcs()
    918 self.compute_pers()
    920 if self.gui:

File ~/.local/lib/python3.10/site-packages/Pyriod/Pyriod.py:1020, in Pyriod._update_lcs(self)
   1018 meanflux = np.mean(self.lc.flux.value[self.include])
   1019 if self.gui:
-> 1020     self.lc_model_sampled.flux = meanflux + self.sample_model(self.lc_model_sampled.time.value)
   1021 #Observed is at all original times (apply mask before calculations)
   1022 self.lc["model"] = (meanflux + self.sample_model(self.lc.time.value))*self.lc.flux.unit

File ~/.local/lib/python3.10/site-packages/astropy/utils/masked/core.py:674, in MaskedNDArray.__array_ufunc__(self, ufunc, method, *inputs, **kwargs)
    671 if m is None:
    672     # TODO: allow writing to unmasked output if nothing is masked?
    673     if d is not None:
--> 674         raise TypeError('cannot write to unmasked output')
    675 elif out_mask is None:
    676     out_mask = m

TypeError: cannot write to unmasked output
```
This is astropy 5.1, Pyriod 0.1.11, Jupyter 1.0.0 and Python 3.10.5. I guess this is a different issue.

**Comment by keatonb** [comment]
@warrickball That disturbing, I can't reproduce that error.  Could be related to another bug I'm fixing.  If you have a chance, could you try again, but do `lc = lc.remove_nans()` before you pass to Pyriod.  Are you trying this with TESS/Kepler data?

**Comment by warrickball** [comment]
I'm using TESS data for SX Phe and have already removed NaNs. It's in essence the same as the script above, though in a Jupyter notebook:
```python
import lightkurve as lk
from Pyriod import Pyriod

lc = lk.search_lightcurve('SX Phe', mission='TESS', exptime=120).download_all()[0].remove_nans()
pyriod = Pyriod(lc)
pyriod.Periodogram()
```
The periodogram starts with the tallest peak select so I click "Add to solution" and then "Refine fit" to produce the error.

**Comment by keatonb** [comment]
Thank you Warrick, I was back to reproduce the issue with that dataset.  I'll get to the bottom of it.

**Comment by keatonb** [comment]
Should really be fixed now.  Thanks again!

**Comment by warrickball** [comment]
I can add to the solution and refine the fit but the `Signals()` and `Pyriod()` cells remain empty... The `Log()` shows correctly, though.

**Comment by keatonb** [comment]
Did you activate the required Jupyter notebook extensions (and restart the Jupyter server after)? https://github.com/keatonb/Pyriod/blob/master/README.md

In the Terminal:
```
jupyter nbextension enable --py --sys-prefix qgrid
jupyter nbextension enable --py --sys-prefix widgetsnbextension
jupyter nbextension enable --py --sys-prefix ipympl
```

**Comment by warrickball** [comment]
Thanks, that seems to have done it! For the record, I'm not running in a virtual environment (I mostly use `python3 -m pip install ... --user`) so I activated the extensions with
```
jupyter nbextension enable --py qgrid --user
jupyter nbextension enable --py widgetsnbextension --user
jupyter nbextension enable --py ipympl --user
```

**Comment by warrickball** [comment]
I do feel kind of bad but I've now got a new error from trying a Kepler target. This is basically the same script as the TESS/SX Phe example but with 16 Cyg A from Kepler:
```python
import lightkurve as lk
from Pyriod import Pyriod

lc = lk.search_lightcurve('16 Cyg A', mission='kepler', exptime=60).download_all()[0].remove_nans()
pyriod = Pyriod(lc)
pyriod.Periodogram()
```
The error is
```
---------------------------------------------------------------------------
UnitTypeError                             Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 pyriod = Pyriod(lc)

File ~/.local/lib/python3.10/site-packages/Pyriod/Pyriod.py:210, in Pyriod.__init__(self, lc, amp_unit, freq_unit, gui, **kwargs)
    207 self._calc_tshift()
    209 # Store version sampled as the data as lc column
--> 210 initmodel = np.zeros(len(self.lc))+np.mean(self.lc.flux[self.include])
    211 self.lc["model"] = initmodel
    212 # Also plot the model over the time series

File ~/.local/lib/python3.10/site-packages/astropy/units/quantity.py:600, in Quantity.__array_ufunc__(self, function, method, *inputs, **kwargs)
    598 if function.nout == 1:
    599     out = out[0]
--> 600 out_array = check_output(out, unit, inputs, function=function)
    601 # Ensure output argument remains a tuple.
    602 kwargs['out'] = (out_array,) if function.nout == 1 else out_array

File ~/.local/lib/python3.10/site-packages/astropy/units/quantity_helper/converters.py:357, in check_output(output, unit, inputs, function)
    354 else:
    355     # output is not a Quantity, so cannot obtain a unit.
    356     if not (unit is None or unit is dimensionless_unscaled):
--> 357         raise UnitTypeError("Cannot store quantity with dimension "
    358                             "{}in a non-Quantity instance."
    359                             .format("" if function is None else
    360                                     "resulting from {} function "
    361                                     .format(function.__name__)))
    363     return output

UnitTypeError: Cannot store quantity with dimension resulting from add function in a non-Quantity instance.
```
At least once I got the `TypeError: cannot write to unmasked output` message above but I can't reproduce that now, after I've killed and restarted the Jupyter kernel several times.

**Comment by keatonb** [comment]
So sorry for all the hassles.  That problem should be fixed now with v0.1.13.  Can't thank you enough for documenting these errors!

**Comment by warrickball** [comment]
Thank you for your support! Everything seems to be working now. Got it crunching on a short-cadence light curve from Kepler. :slightly_smiling_face:
